core/divelist.c

```diff
@@ -85,6 +85,38 @@ static int active_o2(const struct dive *dive, const struct divecomputer *dc, dur
 	return get_o2(gas);
 }
 
+static int get_sample_o2(const struct dive *dive, const struct divecomputer *dc, const struct sample *sample)
+{
+	int po2i, po2f, po2;
+	bool trueo2 = false;
+	struct sample *psample = sample - 1;
```

Comment:
could this ever be called with the very first sample? which would make this an illegal reference?

Reply:
It could but it is not. Added comment to this effect.